File: Class03/app/src/main/java/br/edu/ifsp/souza/charles/pdm/views/MainActivity.java
package br.edu.ifsp.souza.charles.pdm.views;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.TextView;
import android.widget.Toast;

import br.edu.ifsp.souza.charles.pdm.views.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    // Referência para classe de viewBinding
    private ActivityMainBinding amb;
    private Pessoa pessoa;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        amb = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(amb.getRoot());


        amb.salvarBt.setOnClickListener( view -> {
            pessoa = new Pessoa(
                    amb.nomeEt.getText().toString(),
                    amb.sobrenomeEt.getText().toString(),
                    amb.emailEt.getText().toString(),
                    ((TextView) amb.estadoCivilSp.getSelectedView()).getText().toString(),
                    amb.femininoRb.isChecked() ? getString(R.string.feminino) : amb.masculinoRb.getText().toString()
            );

            Toast.makeText(this, pessoa.toString(), Toast.LENGTH_SHORT).show();
        });

        amb.limparBt.setOnClickListener( view -> {
            amb.nomeEt.setText("");
            amb.sobrenomeEt.setText("");
            amb.emailEt.setText("");
            amb.estadoCivilSp.setSelection(0);
            amb.femininoRb.setChecked(true);
            pessoa = null;
        });

        amb.estadoCivilSp.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> adapterView, View view, int posicao, long l) {
                // Caso seja selecinada a opção "casado"
                if (posicao == 1) {
                    amb.conjugeLl.setVisibility(View.VISIBLE);
                } else {
                    amb.conjugeLl.setVisibility(View.GONE);
                }
            }

            @Override
            public void onNothingSelected(AdapterView<?> adapterView) {

            }
        });

    }
    public void onClickBotao(View botao) {
        if (botao.getId() == R.id.salvarBt) {
            Toast.makeText(this, "Clicou em Salvar", Toast.LENGTH_SHORT).show();
        } else {
            Toast.makeText(this, "Clicou no Limpar", Toast.LENGTH_SHORT).show();
        }
    }
}

File: Class09/app/src/main/java/br/edu/ifsp/souza/charles/pdm/mycontacts/MainActivity.kt
package br.edu.ifsp.souza.charles.pdm.mycontacts

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.ArrayAdapter
import br.edu.ifsp.souza.charles.pdm.mycontacts.databinding.ActivityMainBinding
import br.edu.ifsp.souza.charles.pdm.mycontacts.model.Contact

class MainActivity : AppCompatActivity() {

    private val amb: ActivityMainBinding by lazy {
        ActivityMainBinding.inflate(layoutInflater)
    }
    // Data source
    private val contactList: MutableList<Contact> = mutableListOf()

    //Adaptar
    private val contactAdpater: ArrayAdapter<String> by lazy {
        ArrayAdapter(
            this,
            android.R.layout.simple_list_item_1,
            contactList.map{ it.name}
        )
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(amb.root)

        fillContactList()
    }

    private fun fillContactList(){
        for ( i in 1 .. 50) {
            contactList.add(
                Contact(
                    id = 1,
                    name = "Nome $1",
                    address = "Endereço $1",
                    phone = "Telefone $1",
                    email = "Email $1",
                )
            )
        }
    }
}